app/index.js

```diff
@@ -39,6 +39,29 @@ app.on('window-all-closed', () => {
   }
 });
 
+let quitIsConfirmed = false;
+
+app.on('before-quit', (event) => {
+  const windows = BrowserWindow.getAllWindows().filter(win => win.sessions);
```

Comment:
I believe we're keeping a `Set` of windows (a copy of which we return from `getWindows`). Maybe we don't need `getAllWindows()`?